scenario_player/runner.py

```diff
@@ -153,279 +256,260 @@ def local_seed(self) -> str:
+        user_deposit_proxy, token_proxy = get_udc_and_corresponding_token(
+            udc_settings=udc_settings, settings=settings, proxy_manager=self.proxy_manager
         )
-        greenlets.add(setup)
 
-        gevent.joinall(greenlets, raise_error=True, count=1)
+        allowance_tx = userdeposit_maybe_increase_allowance(
+            token_proxy, user_deposit_proxy, required_allowance, maximum_allowance=UINT256_MAX
+        )
+        if allowance_tx:
+            orchestration_user_deposit_token_transactions.add(allowance_tx)
 
-    def _initialize_scenario_token(
-        self,
-        node_addresses: Set[ChecksumAddress],
-        udc_ctr: Optional[Contract],
-        should_deposit_ud_token: bool,
-        gas_limit: int,
+        mint_tx = token_maybe_mint(
+            token_proxy,
+            to_checksum_address(self.client.address),
+            minimum_balance=required_allowance,
+            maximum_balance=UINT256_MAX,
+        )
+        if mint_tx:
+            orchestration_user_deposit_token_transactions.add(mint_tx)
+
+        return orchestration_user_deposit_token_transactions
+
+    def setup_raiden_token_balances(
+        self, token_proxy: Token, node_addresses: Set[ChecksumAddress]
     ) -> Set[TransactionHash]:
-        """Create or reuse an existing token, and mint the token for every
-        `node_addresses`.
+        """Mint the necessary amount of tokens from `token_proxy` for every `node_addresses`.
+
+        This will use the scenario player's account, therefore it doesn't have
+        to wait for the ether transfers to finish.
         """
-        self.token.init()
-        mint_tx = set()
-        for address in node_addresses:
-            tx = self.token.mint(address)
-            if tx:
-                mint_tx.add(tx)
+        transaction_hashes = set()
+        token_min_amount = self.config.token.min_balance
+        token_max_amount = self.config.token.max_funding
 
-            if not should_deposit_ud_token:
-                continue
-            assert self.udc
-            deposit_tx = self.udc.deposit(address)
-            if deposit_tx:
-                mint_tx.add(deposit_tx)
+        for address in node_addresses:
+            mint_txhash = token_maybe_mint(
+                token=token_proxy,
+                target_address=address,
+                minimum_balance=token_min_amount,
+                maximum_balance=token_max_amount,
+            )
+            if mint_txhash:
+                transaction_hashes.add(mint_txhash)
 
-        return mint_tx
+        return transaction_hashes
 
-    def _initialize_udc(
-        self, gas_limit: int, node_count: int
-    ) -> Tuple[Set[TransactionHash], Optional[Contract], bool]:
-        our_address = to_checksum_address(self.client.address)
-        udc_settings = self.definition.settings.services.udc
-        udc_enabled = udc_settings.enable
+    def setup_raiden_nodes_with_sufficient_user_deposit_balances(
```

Comment:
This used to be part of the [`_initialize_scenario_token`](https://github.com/raiden-network/scenario-player/blob/f208ed6cb4f713c1efc921b10458365bcf918b5d/scenario_player/runner.py#L325-L330), which did minting for both the user deposity utility token and for the token network token. Now split.